modules/foreman_job_template.py

```diff
@@ -0,0 +1,522 @@
+    verify_ssl = entity_dict.pop('verify_ssl')
+    state = entity_dict.pop('state')
+    file_name = entity_dict.pop('file_name', None)
+
+    if file_name or 'template' in entity_dict:
+        if file_name:
+            parsed_dict = parse_template_from_file(file_name, module)
+        else:
+            parsed_dict = parse_template(entity_dict['template'], module)
+        # sanitize name from template data
+        # The following condition can actually be hit, when someone is trying to import a
+        # template with the name set to '*'.
+        # Besides not being sensible, this would go horribly wrong in this module.
+        if 'name' in parsed_dict and parsed_dict['name'] == '*':
+            module.fail_json(msg="Cannot use '*' as a job template name!")
+        # module params are priorized
+        parsed_dict.update(entity_dict)
+        entity_dict = parsed_dict
+
+    # make sure, we have a name
+    if 'name' not in entity_dict:
+        if file_name:
+            entity_dict['name'] = os.path.splitext(
+                os.path.basename(file_name))[0]
+        else:
+            module.fail_json(
+                msg='No name specified and no filename to infer it.')
+
+    name = entity_dict['name']
+
+    affects_multiple = name == '*'
+    # sanitize user input, filter unuseful configuration combinations with 'name: *'
+    if affects_multiple:
+        if state == 'present_with_defaults':
+            module.fail_json(msg="'state: present_with_defaults' and 'name: *' cannot be used together")
+        if state == 'absent':
+            if entity_dict.keys() != ['name']:
+                module.fail_json(msg="When deleting all job templates, there is no need to specify further parameters.")
+
+    try:
+        create_server(server_url, (username, password), verify_ssl)
+    except Exception as e:
+        module.fail_json(msg='Failed to connect to Foreman server: %s ' % e)
+
+    ping_server(module)
+
+    try:
+        if affects_multiple:
+            entities = find_entities(JobTemplate)
+        else:
+            entities = find_entities(JobTemplate, name=entity_dict['name'])
+    except Exception as e:
+        module.fail_json(msg='Failed to find entity: %s ' % e)
+
+    # Set Locations of job template
+    if 'locations' in entity_dict:
+        entity_dict['locations'] = find_entities_by_name(
+            Location, entity_dict['locations'], module)
+
+    # Set Organizations of job template
+    if 'organizations' in entity_dict:
+        entity_dict['organizations'] = find_entities_by_name(
+            Organization, entity_dict['organizations'], module)
+
+    # TemplateInputs need to be added as separate entities later
+    if 'template_inputs' in entity_dict:
+        template_input_list = entity_dict['template_inputs']
+
+    entity_dict = sanitize_entity_dict(entity_dict, name_map)
+
+    changed = False
+    if not affects_multiple:
+        if len(entities) == 0:
```

Comment:
Generally I prefer to avoid len for these cases:
```python
if entities:
    entity = entities[0]
else:
    entity = None
```

Alternative is obviously to catch the `IndexError`